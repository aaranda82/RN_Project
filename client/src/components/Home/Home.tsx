import React from 'react';
import {
  Button,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import {HomeProps} from '../../Types';

const Home: React.FC<HomeProps> = ({navigation}) => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };
  return (
    <>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
          }}>
          <Text style={[styles.sectionTitle, styles.topText]}>
            Super Sweet Mobile App
          </Text>
          <Text style={styles.subTitle}>Create an account and log in</Text>
          <Text style={styles.subTitle}>
            to view all the <Text style={styles.highlight}>super sweet</Text>{' '}
            content
          </Text>
          <Button
            title="Login"
            onPress={() => {
              navigation.navigate('Login');
            }}
          />
          <Button
            title="Sign up"
            onPress={() => {
              navigation.navigate('Register');
            }}
          />
        </View>
      </ScrollView>
    </>
  );
};

const styles = StyleSheet.create({
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  topText: {textAlign: 'center', margin: 30},
  subTitle: {textAlign: 'center', fontSize: 16, color: '#566573'},
  highlight: {fontWeight: '600'},
});

export default Home;
