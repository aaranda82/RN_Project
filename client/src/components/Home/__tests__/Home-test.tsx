import React from 'react';
import Home from '../Home';

import renderer from 'react-test-renderer';

const navProps: any = {};
const routeProps: any = {};

it('renders Home screen correctly', () => {
  const tree = renderer
    .create(<Home navigation={navProps} route={routeProps} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
