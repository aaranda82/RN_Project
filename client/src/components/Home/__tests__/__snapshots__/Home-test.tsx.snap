// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home screen correctly 1`] = `
<RCTScrollView
  contentInsetAdjustmentBehavior="automatic"
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "600",
            },
            Object {
              "margin": 30,
              "textAlign": "center",
            },
          ]
        }
      >
        Super Sweet Mobile App
      </Text>
      <Text
        style={
          Object {
            "color": "#566573",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Create an account and log in
      </Text>
      <Text
        style={
          Object {
            "color": "#566573",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        to view all the 
        <Text
          style={
            Object {
              "fontWeight": "600",
            }
          }
        >
          super sweet
        </Text>
         
        content
      </Text>
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign up
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
